jme3-core/src/main/java/com/jme3/anim/tween/action/BlendAction.java

```diff
@@ -1,129 +1,265 @@
+/*
+ * Copyright (c) 2009-2021 jMonkeyEngine
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 package com.jme3.anim.tween.action;
 
 import com.jme3.anim.util.HasLocalTransform;
 import com.jme3.math.Transform;
-
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * A concrete implementation used to blend between 2 actions #{@link BlendAction#firstActiveIndex}, #{@link BlendAction#secondActiveIndex} & apply them on their targets,
+ * blending between 2 actions is done by interpolating the set of transforms serially starting from the firstAction transforms, moving on to the second one transforms,
+ * using the blendWeight as a delta value, then interpolating the resulted value with the current target local transforms,
+ * and finalizing by applying the final collected result of interpolation to the target as a local transform matrix.
+ *
+ * If the blendWeight is :
+ * - less than 1, then interpolate the 1st action & the second one & apply the results.
+ * - equal to zero, then interpolate the 1st action only & apply the results.
+ * - equal to one, then interpolate only the second action with blendWeight = 1.
+ *
+ * <b>Created by Nehon.</b>
+ */
 public class BlendAction extends BlendableAction {
 
     private int firstActiveIndex;
     private int secondActiveIndex;
     final private BlendSpace blendSpace;
-    private float blendWeight;
-    final private double[] timeFactor;
+    private double[] timeFactor;
```

Comment:
Why did you delete `blendWeight` here? Why did remove the `final` qualifier from `timeFactor`?

Reply:
The global field `blendWeight` here is used only at one place so no need to be global then.

The `timeFactor` field was final as it was used inside the constructor directly, but when moving what's inside the constructor to relevant methods, the `final` needs to be removed.

Reply:
Code changes shouldn't be included in a documentation PR. Please revert those changes. You can submit them in the future as part of a different PR.

Reply:
This won't be shipped on the new PR, no worries.